kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/FibonacciIntervalFn.kt

```diff
@@ -0,0 +1,47 @@
+package io.kotest.assertions.nondeterministic
+
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.milliseconds
+
+/**
+ * Fibonacci delay implements a delay where each duration is calculated as a multiplier
+ * of the fibonacci sequence, 0, 1, 1, 2, 3, 5....
+ *
+ * Some people start fib at 0, some at 1.
+ * This implementation starts with 0 as per https://en.wikipedia.org/wiki/Fibonacci_number
+ *
+ * @param offset   Added to the count, so if the offset is 4, then the first value will be the 4th fib number.
+ * @param base The duration that is multiplied by the fibonacci value
+ * @param max the maximum duration to clamp the resulting duration to defaults to [FibonacciIntervalFn.defaultMax]
+ */
+class FibonacciIntervalFn(private val base: Duration, private val offset: Int, private val max: Duration?) : DurationFn {
```

Comment:
```suggestion
@KotestInternal
class FibonacciIntervalFn(private val base: Duration, private val offset: Int, private val max: Duration?) : DurationFn {
```

I dont think we want users to instantiate this directly?